paddle/fluid/operators/expand_as_op.h

```diff
@@ -32,10 +32,10 @@ limitations under the License. */
   }
 #define REP_EXPAND_AS_TEMPLATE(n) BOOST_PP_REPEAT(n, EXPAND_AS_TEMPLATE, ~)
 #define COND(n) BOOST_PP_GREATER_EQUAL(n, BOOST_PP_MOD(n, MAX_RANK_SUPPORTED))
-#define EXPAND_AS_GRAD_CASE(n)                                       \
-  case n: {                                                          \
-    ExpandAsBackward<n>(context, reshape_dims_vec, reduce_dims_vec); \
-    break;                                                           \
+#define EXPAND_AS_GRAD_CASE(n)                                           \
+  case n + 1: {                                                          \
```

Comment:
just change from n to n+1 ?

so the new EXPAND_AS_GRAD_CASE(1) is equal to the original EXPAND_AS_GRAD_CASE(2)?
but the original EXPAND_AS_GRAD_CASE(1) is removed?

Reply:
actually, the original EXPAND_AS_GRAD_CASE(2) is wrong. it generate case 1:xxxxxx
the original EXPAND_AS_GRAD_CASE generate case 0-5.
by change n->n+1, EXPAND_AS_GRAD_CASE generate case 1-6.
finally, the switch case code should be:
case 1:
xxxxxx
case 2:
xxxxxx
... ...
case 6:
xxxxxx

and case 0 is not allowed, before the switch case 
PADDLE_ENFORCE_GE(dims, 1,
                            "expand_as_v2_grad op must be greater than or "
                            "equal to 1, but the value received is %d.",
                            dims));

Reply:
already add comments and switch case default.